workers/unity/Packages/com.improbable.gdk.core/Tests/Editmode/Systems/SpatialOSReceiveSystemTests.cs

```diff
@@ -0,0 +1,738 @@
+            var entityQueryRequestsHandle =
+                entityManager.GetComponentData<WorldCommands.EntityQuery.CommandSender>(worker.WorkerEntity).Handle;
+            var reserveEntityIdsRequestsHandle =
+                entityManager.GetComponentData<WorldCommands.ReserveEntityIds.CommandSender>(worker.WorkerEntity)
+                    .Handle;
+
+            using (var wrappedOp = WorkerOpFactory.CreateDisconnectOp(DisconnectReason))
+            {
+                receiveSystem.OnDisconnect(wrappedOp.Op);
+            }
+
+            Assert.IsTrue(entityManager.HasComponent<OnDisconnected>(worker.WorkerEntity));
+            Assert.AreEqual(DisconnectReason,
+                entityManager.GetSharedComponentData<OnDisconnected>(worker.WorkerEntity).ReasonForDisconnect);
+
+            Assert.Throws<ArgumentException>(() =>
+                WorldCommands.CreateEntity.RequestsProvider.Get(createEntityRequestsHandle));
+            Assert.Throws<ArgumentException>(() =>
+                WorldCommands.DeleteEntity.RequestsProvider.Get(deleteEntityRequestsHandle));
+            Assert.Throws<ArgumentException>(() =>
+                WorldCommands.EntityQuery.RequestsProvider.Get(entityQueryRequestsHandle));
+            Assert.Throws<ArgumentException>(() =>
+                WorldCommands.ReserveEntityIds.RequestsProvider.Get(reserveEntityIdsRequestsHandle));
+        }
+
+        [Test]
+        public void OnCreateEntityResponse_should_add_received_responses_to_entity()
+        {
+            var entity = SetupTestEntity();
+
+            var emptyRequest = new WorldCommands.CreateEntity.Request();
+            var context = "Some context";
+
+            createEntityStorage.CommandRequestsInFlight.Add(TestCommandRequestId,
+                new CommandRequestStore<WorldCommands.CreateEntity.Request>(entity,
+                    emptyRequest, context, TestCommandRequestId));
+
+            using (var wrappedOp = WorkerOpFactory.CreateCreateEntityResponseOp(TestCommandRequestId))
+            {
+                receiveSystem.OnCreateEntityResponse(wrappedOp.Op);
+
+                Assert.IsTrue(entityManager.HasComponent<WorldCommands.CreateEntity.CommandResponses>(entity));
+
+                var responses = entityManager.GetComponentData<WorldCommands.CreateEntity.CommandResponses>(entity);
+
+                var count = 0;
+                Assert.DoesNotThrow(() => { count = responses.Responses.Count; });
+                Assert.AreEqual(1, count);
+
+                var response = responses.Responses[0];
+
+                Assert.AreEqual(emptyRequest, response.RequestPayload);
+                Assert.AreEqual(context, response.Context);
+                Assert.AreEqual(TestCommandRequestId, response.RequestId);
+                Assert.AreEqual(wrappedOp.Op, response.Op);
+            }
+        }
+
+        [Test]
+        public void OnCreateEntityRespoonse_should_error_if_request_id_not_found()
```

Comment:
typo: `OnCreateEntityRespoonse` ->  `OnCreateEntityResponse`

Reply:
Its actually on receiving a create entity spoons this gets triggered - Hence `OnCreateEntityRe(ceiving)spoons`. The trailing e is the real typo

Reply:
🥄

Reply:
ah, of course. entity spoons, how could I have misread that xD